server/data/projects.js

```diff
@@ -93,7 +97,57 @@ const createProject = async (projectObjParam, user) => {
 	return createdProject;
 };
 
+const updateProject = async (projectObjParam, id, user) => {
+	let projectCheck = await getProjectById(id);
+	ownercheck = checkuseraccess(user, projectCheck.owner);
+	if (!ownercheck)
+		throw forbiddenErr(
+			`Not Authorised to update this project. Not Project Owner`
+		);
+	const projectObj = isValidProjectObject(projectObjParam);
+	const { name, description, github, media, technologies, deploymentLink } =
+		projectObj;
+	const date = new Date();
+	const updateProjectObject = {
```

Comment:
this is an incomplete object, many keys are not here. this will overwrite the entire object in the DB. i think you can use object spreading here, something like:
```javascript
const updateProjectObject = {
    ...updateProject,
    name,
    other keys that are updated
}
```